src/web/lib/storage.js

```diff
@@ -17,16 +17,26 @@ const generateThemeKey = () =>
   `${Date.now()}-${Math.floor(Math.random() * 1000)}`;
 
 function putTheme(key, theme) {
-  log("putTheme", key, theme);
-  const storageKey = themeStorageKey(key);
-  localStorage.setItem(
-    storageKey,
-    JSON.stringify({
-      theme,
-      modified: Date.now()
-    })
-  );
-  notifySelfForStorage(storageKey);
+  let isThemeDuplicate = checkDuplicateTheme(theme);
+  if (!isThemeDuplicate) {
+    log("putTheme", key, theme);
+    const storageKey = themeStorageKey(key);
+    localStorage.setItem(
+      storageKey,
+      JSON.stringify({
+        theme,
+        modified: Date.now()
+      })
+    );
+    notifySelfForStorage(storageKey);
+  }
+}
+
+function checkDuplicateTheme(theme) {
+  const themesList = listThemes();
+  return Object.keys(listThemes()).some((key) => {
```

Comment:
Sorry for being annoying...
Small nit, the `listThemes()` here should be `themesList` from L36 above, so we aren't calling the method twice.

Reply:
Ahh, I missed that one. Thanks. Will push the changes.